qp_shotgun/qc_trim/qc_trim.py

```diff
@@ -0,0 +1,266 @@
+            dash = '-'
+        if value is 'True':
+            params.append('%s%s' % (dash, param))
+        elif value is 'False':
+            continue
+        elif value and value != 'default':
+            params.append('%s%s %s' % (dash, param, value))
+
+    param_string = ' '.join(params)
+
+    return(param_string)
+
+
+def generate_qc_trim_commands(forward_seqs, reverse_seqs, map_file,
+                              out_dir, parameters):
+    """Generates the QC_Trim commands
+
+    Parameters
+    ----------
+    forward_seqs : list of str
+        The list of forward seqs filepaths
+    reverse_seqs : list of str
+        The list of reverse seqs filepaths
+    map_file : str
+        The path to the mapping file
+    out_dir : str
+        The job output directory
+    parameters : dict
+        The command's parameters, keyed by parameter name
+
+    Returns
+    -------
+    cmds: list of str
+        The KneadData commands
+    samples: list of tup
+        list of 4-tuples with run prefix, sample name, fwd read fp, rev read fp
+
+    Notes
+    -----
+    Currently this is requiring matched pairs in the make_read_pairs_per_sample
+    step but implicitly allowing empty reverse reads in the actual command
+    generation. This behavior may allow support of situations with empty
+    reverse reads in some samples, for example after trimming and QC.
+    """
+    # we match filenames, samples, and run prefixes
+    samples = make_read_pairs_per_sample(forward_seqs, reverse_seqs, map_file)
+    cmds = []
+
+    param_string = _format_qc_trim_params(parameters)
+
+    for run_prefix, sample, f_fp, r_fp in samples:
+        cmds.append('atropos trim --threads 4 %s -o %s -p %s -pe1 %s -pe2 %s'
+                    % (param_string, join(out_dir, '%s.R1.trimmed.fastq.gz' %
+                       sample), join(out_dir, '%s.R2.trimmed.fastq.gz' %
+                       sample), f_fp, r_fp))
+    return cmds, samples
+
+
+def _run_commands(qclient, job_id, commands, msg):
+    for i, cmd in enumerate(commands):
+        qclient.update_job_step(job_id, msg % i)
+        std_out, std_err, return_value = system_call(cmd)
+        if return_value != 0:
+            error_msg = ("Error running QC_Trim:\nStd out: %s\nStd err: %s"
+                         "\n\nCommand run was:\n%s"
+                         % (std_out, std_err, cmd))
+            return False, error_msg
+
+    return True, ""
+
+
+def _per_sample_ainfo(out_dir, samples, fwd_and_rev=False):
+    files = []
+    missing_files = []
+
+    suffixes = ['%s.R1.trimmed.fastq.gz', '%s.R2.trimmed.fastq.gz']
+
+    for _, rp, _, _ in samples:
+        smd = partial(join, out_dir)
+        for suff in suffixes:
+            fname = smd(suff % rp)
+            if exists(fname):
+                files.append(fname)
+            else:
+                missing_files.append(fname)
+
+    if not files:
+        # KneadData did not create any files, which means that no sequence
```

Comment:
We don't have KneadData anymore ... is this outdated?

Reply:
No, we essentially copied the kneadData file and changed the functions within. I have not update the documentation yet.